llama_stack/apis/models/models.py

```diff
@@ -7,20 +7,38 @@
 from typing import Any, Dict, List, Literal, Optional, Protocol, runtime_checkable
 
 from llama_models.schema_utils import json_schema_type, webmethod
-from pydantic import Field
+from pydantic import BaseModel, Field
 
 from llama_stack.apis.resource import Resource, ResourceType
 
 
-@json_schema_type
-class Model(Resource):
-    type: Literal[ResourceType.model.value] = ResourceType.model.value
+class CommonModelFields(BaseModel):
     metadata: Dict[str, Any] = Field(
         default_factory=dict,
         description="Any additional metadata for this model",
     )
 
 
+@json_schema_type
+class Model(CommonModelFields, Resource):
+    type: Literal[ResourceType.model.value] = ResourceType.model.value
+
+    @property
+    def model_id(self) -> str:
+        return self.identifier
+
+    @property
+    def provider_model_id(self) -> str:
+        return self.provider_resource_id
+
+
+@json_schema_type
+class ModelInput(CommonModelFields):
```

Comment:
Wondering if the `register_model` should also use `ModelInput`?

```python
async def register_model(
        self,
        model_id: str,
        provider_model_id: Optional[str] = None,
        provider_id: Optional[str] = None,
        metadata: Optional[Dict[str, Any]] = None,
    ) -> Model: ...
```

If seems like we have 3 different structures now associated with a resource?
- `Model` --> datastrcuture for model fields
- `ModelInput` --> for specifying StackRunConfig pre-registration
- `register_model` --> flattened args for `ModelInput`?

Reply:
That is correct. For now, I left it as now because I think register_model()'s API looks nice as is. Adding a structure to it makes it feel unnecessarily annoying. And internally, ModelInput does get destructured and passed to register_model() so any incompatibility is immediately flagged.